src/Raven.Server/Documents/Indexes/Index.cs

```diff
@@ -2426,7 +2443,27 @@ private void DisposeIndexWriterOnError(Lazy<IndexWriteOperationBase> writeOperat
                     writeOperation.Value.Dispose();
             }
         }
+        
+        private Dictionary<string, bool> _collectionTracker;
+        
+        public IIndexedItemEnumerator GetMapEnumeratorAndBoxWhenNeeded(IEnumerable<IndexItem> items, string collection, TransactionOperationContext indexContext,
```

Comment:
What is the meaning of `AndBoxWhenNeeded` from the caller point of view? Is that important for him? What does "box" mean actually?

Wrapping the enumerator is transparent for the caller anyway, so maybe `EnumerateIndexedItems()` would be more suitable name?

Reply:
I renamed it to `EnumerateIndexedItems`